pkg/scheduler/framework/preemption/preemption.go

```diff
@@ -170,7 +170,7 @@ func (ev *Evaluator) Preempt(ctx context.Context, pod *v1.Pod, m framework.NodeT
 			},
 		}
 		// Specify nominatedNodeName to clear the pod's nominatedNodeName status, if applicable.
-		return framework.NewPostFilterResultWithNominatedNode(""), framework.NewStatus(framework.Unschedulable, fitError.Error())
+		return framework.NewPostFilterResultWithNominatedNode(""), framework.NewStatus(framework.Unschedulable).WithError(fitError)
```

Comment:
> we assume it was not an error status?

Nope, a status's category is determined by its `Code`, so here it's `Unschedulable` instead of `Error`.

> now it is set to fitError, should this be tested?

It's covered by existing tests. Now when composing a Status, we delay the evaluation of embedded `err` to the time of invoking `status.Reasons()`.